pkg/kubelet/lifecycle/predicate.go

```diff
@@ -126,11 +130,11 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		// If there are failed predicates, we only return the first one as a reason.
 		r := reasons[0]
 		switch re := r.(type) {
-		case *predicates.PredicateFailureError:
+		case *PredicateFailureError:
```

Comment:
Would this cause problems during kubernetes upgrade? For example, if update a kubernetes cluste from 1.17 to 1.18. Some kubelet are still using the 1.17 version while the kube-scheduler is using 1.18.

Reply:
Nope. This was a library-level dependency, not runtime dependency (like API). The dependency was sealed at the moment when the old-version kubelet binary was cut.